test/unit/utils/Client.util.test.ts

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2022 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import {getGRPCMetadata, getHTTPMetadataQueryParam} from "../../../src/utils/Client.util"
+
+describe('Client.util', () => {
```

Comment:
* Also add when metadata is null or undefined check
* Also add hashbang == '#' (e.g. http://www.facebook.com/example.profile#!/pages/Another-Page/123456789012345)